### PR TITLE
Updates to VDI Optimizations for Windows 10 2004

### PR DESCRIPTION
**Pull request by msft-jasonparker**
Verified these settings work through manual and automated deployments.  Please contact to discuss in detail.

### CODE REVIEW

File: Win10_VirtualDesktop_Optimize.ps1 [hunk 1]
```diff
@@ -26,16 +26,37 @@
 ######################################################################################################################################>
 
 [Cmdletbinding()]
-Param
-(
-    [Parameter()]
-    [ValidateSet('1909','2004')]
-    $WindowsVersion = 2004,
+Param (
+    # Parameter help description
```
Comment (tmmuessig) [inline Win10_VirtualDesktop_Optimize.ps1:30]: Love that all of the tasks are broken out, but we need a way with one switch that will run them all, possibly in a second parameter set

File: Win10_VirtualDesktop_Optimize.ps1 [hunk 2]
```diff
@@ -82,114 +103,205 @@ The Store and a few others, such as Wallet, were left off intentionally.  Though
 it is nearly impossible to get it back.  Please review the lists below and comment out or remove references to packages that you do not want to remove.
 #>
 
+$VerbosePreference = "Continue"
+$WindowsVersion = (Get-ItemProperty "HKLM:\Software\Microsoft\Windows NT\CurrentVersion\").ReleaseId
```
Comment (tmmuessig) [inline Win10_VirtualDesktop_Optimize.ps1:107]: We need to be able to deviate from the specific version of windows.  This is pointing them to the configuration folder, but does not allow for a choice of a custom folder where they might have made their own Config files

File: Win10_VirtualDesktop_Optimize.ps1 [hunk 3]
```diff
@@ -82,114 +103,205 @@ The Store and a few others, such as Wallet, were left off intentionally.  Though
...
-        #$EnabledScheduledTasks | Where-Object { $_.TaskName -like "*$Task*" } #| Disable-ScheduledTask
+If ($ScheduledTasks) {
+    If (Test-Path .\ConfigurationFiles\ScheduledTasks.json) {
+        Write-Output ("[VDI Optimize] Disable Scheduled Tasks")
+        $SchTasksList = Get-Content .\ConfigurationFiles\ScheduledTasks.json | ConvertFrom-Json | Where-Object { $_.VDIState -eq 'Disabled' }
+        If ($SchTasksList.count -gt 0) {
+            Foreach ($Item in $SchTasksList) {
+                $TaskObject = Get-ScheduledTask $Item.ScheduledTask
+                If ($TaskObject -and $TaskObject.State -ne 'Disabled') {
+                    Write-Verbose ("Attempting to disable Scheduled Task: {0}" - $TaskObject.TaskName)
```
Comment (tmmuessig) [inline Win10_VirtualDesktop_Optimize.ps1:175]: missing the "-**F**" in the command

**Review by tmmuessig — CHANGES_REQUESTED** [review]

File: Win10_VirtualDesktop_Optimize.ps1 (hunk 1 above)
Comment (msft-jasonparker) [reply Win10_VirtualDesktop_Optimize.ps1:30]: Yeah, I agree.  There should be a default param set for all the tasks vs picking and choosing each task.

File: Win10_VirtualDesktop_Optimize.ps1 (hunk 2 above)
Comment (msft-jasonparker) [reply Win10_VirtualDesktop_Optimize.ps1:107]: Maybe we update the param block and allow the user to select a windows version of their choice, but the default value would be the version pulled back from the registry.  Thoughts?

File: Win10_VirtualDesktop_Optimize.ps1 (hunk 3 above)
Comment (msft-jasonparker) [reply Win10_VirtualDesktop_Optimize.ps1:175]: Yes, I found that yesterday 👎

**Comment by tmmuessig** [comment]
Spoke with author, new pull request is coming with additional changes. Closing this.

**Pull request closed without merging**